src/display/svg.js

```diff
@@ -936,6 +946,91 @@ SVGGraphics = (function SVGGraphicsClosure() {
       this.current.tspan = this.svgFactory.createElement('svg:tspan');
       this.current.xcoords = [];
     },
+    setStrokeColorN: function SVGGraphics_setStrokeColorN(args) {
+      this.current.strokeColor = this._makeColorN_Pattern(args);
+    },
+    setFillColorN: function SVGGraphics_setFillColorN(args) {
+      this.current.fillColor = this._makeColorN_Pattern(args);
+    },
+    shadingFill: function SVGGraphics_shadingFill(args) {
+      var viewport = this.viewport;
+      var width = viewport.width;
+      var height = viewport.height;
+      var inv = Util.inverseTransform(this.transformMatrix);
+      var bl = Util.applyTransform([0, 0], inv);
+      var br = Util.applyTransform([0, height], inv);
+      var ul = Util.applyTransform([width, 0], inv);
+      var ur = Util.applyTransform([width, height], inv);
+      var x0 = Math.min(bl[0], br[0], ul[0], ur[0]);
+      var y0 = Math.min(bl[1], br[1], ul[1], ur[1]);
+      var x1 = Math.max(bl[0], br[0], ul[0], ur[0]);
+      var y1 = Math.max(bl[1], br[1], ul[1], ur[1]);
+
+      var rect = this.svgFactory.createElement('svg:rect');
+      rect.setAttributeNS(null, 'x', x0);
+      rect.setAttributeNS(null, 'y', y0);
+      rect.setAttributeNS(null, 'width', x1 - x0);
+      rect.setAttributeNS(null, 'height', y1 - y0);
+      rect.setAttributeNS(null, 'fill', this._makeShadingPattern(args));
+      this._ensureTransformGroup().appendChild(rect);
+    },
+    _makeColorN_Pattern: function SVGGraphics_makeColorN_Pattern(args) {
+      if (args[0] === 'TilingPattern') {
+        warn('Unimplemented: TilingPattern');
+        return null;
+      }
+      return this._makeShadingPattern(args);
+    },
+    _makeShadingPattern: function SVGGraphics_makeShadingPattern(args) {
+      switch (args[0]) {
+        case 'RadialAxial':
+          var shadingId = 'shading' + shadingCount++;
+          var colorStops = args[2];
+          var gradient = this.svgFactory.createElement('svg:linearGradient');
```

Comment:
In the case of a radial gradient, I think we should use `svg:radialGradient` instead. According to https://developer.mozilla.org/en-US/docs/Web/SVG/Element/linearGradient, there are no `cx` and `cy` attributes, whereas `radialGradient` has those.

Reply:
Thanks very much. I fixed it.
Maybe I mistaked when I refactored for the eslint warnings...